### DISCUSSION TITLE
Full-path links to anchors and headings in exported .html's lead to source .txt's

### DISCUSSION THREAD

**Discussion by Mike221223** [zim-desktop-wiki/zim-desktop-wiki#2487] [Q&A] [answered]
**My configuration**:
Operating System: Linux Mint 21.2
Kernel: Linux 5.15.0-86-generic
Architecture: x86-64
Hardware Vendor: Purism
Hardware Model: Librem Mini v2
Zim version 0.75.2 (deb package)
Browsers: Firefox, Librewolf, Chromium

**The issue**
Links to id's, -- anchors and headings -- when exported to html, lead to source .txt files. Instead of jumping to an id'ed line, my browsers open the .txt.

It appears to only happen when a full path is used. For instance, if I use context menu ("Copy link to this location") to copy a link to a heading that says "Chapter 1", I get text that looks like this: `My_books:One_of_my_books#chapter-1`. Then if I link it to another piece of text in the document, it won't work in the exported html file; again, the link will lead to the original .txt. If, however, I use a short path -- `#chapter-1`, it works as intended in the export.

Same happens with links to anchors. If I link to a footnote, -- `My_books:One_of_my_books#footnote1` -- the link won't work in html. Setting the link to `#footnote1` solves the issue.

In Zim links work with no issues in either case.

Links to higher- and lower-level pages seem to work fine as well, though I haven't done thorough checks.

I really hope it's an issue that can be fixed, or better yet I'm doing something wrong, because, boy oh boy, how many links I've got now to correct in over 300 documents.

**Comment by introt** [comment]
I can't reproduce on the latest `develop` commit. Could you create a new clean notebook with a minimal example and provide the page source files? Furthermore - to make testing easier for everyone involved - is the issue present when using the integrated web server? (Tools > Start web server - it transparently exports the pages as HTML as needed)

**Reply by Mike221223** [reply]
I wrote a page in a clean notebook to test the issue. Here is the source .txt: [Test_page.txt](https://github.com/zim-desktop-wiki/zim-desktop-wiki/files/12841399/Test_page.txt).

I described the testing in the file. Here's the Results section:

### Printing to browser

The issue I described in the opening post of this discussion was reproduced upon using print-to-browser function.

Full-path links (the link to heading 2 and the link to the last line under heading 2) lead to id's in the attached .txt file; in my case, they lead to:

	Test_page.txt#heading-2
	Test_page.txt#line2

respectively.

Short links work as intended. They lead to:

	tmp/zim-5gzq9uat/print-to-browser.html#heading-1
	tmp/zim-5gzq9uat/print-to-browser.html#line1

respectively.

### Exporting to .html

I was incorrect in my description of the issue, attributing it only to the exported .html's. So it appears this issue shows symptoms in both printing-to-browser and exporting to .html.

In the exported .html of the attached file, full-path links are turned into these:

	Test_page.html#Test page#heading-2
	Test_page.html#Test page#line2

They don't work at all, that is, if I click on them, nothing happens.

The short links work as intended. They look like this:

	Test_page.html#heading-1
	Test_page.html#line1

This issue looks similar to the one I posted about a while ago ([⬀](https://github.com/zim-desktop-wiki/zim-desktop-wiki/discussions/2264)). That issue appears to have been fixed with the latest release.

**Reply by introt** [reply, marked as answer]
Thanks for the thorough response!

The bug isn't reproducible using the integrated web server - the links work there. Here's the result of `diff webserver export`:

```diff
74,75c74,75
< A link to heading 2: <a href="/Test_page.html#heading-2" title="⇩" class="page">⇩</a><br>
< A link to the last line (<tt>line2</tt>) under heading 2: <a href="/Test_page.html#line2" title="⇩" class="page">⇩</a>
---
> A link to heading 2: <a href="#Test page#heading-2" title="⇩" class="page">⇩</a><br>
> A link to the last line (<tt>line2</tt>) under heading 2: <a href="#Test page#line2" title="⇩" class="page">⇩</a>
187c187
< <a href='/Test_page.html'>Test page</a>
---
> <a href='#Test page'>Test page</a>
```

I don't see links to txt files, but it seems the single-page export (tested here) refers to the current page as `#Test page` which works fine, but `#Test page#line2` obviously doesn't work, and that's a bug. Would you care to check if the links work on your system when using Tools > Start web server before filing a bug ticket?

**Reply by Mike221223** [reply]
Links do work without issues when browsed on a web server.

I used “Start Web Server” function for the first time, and now I know what it does. Thank you. I'll be using it more. If it matters, I used all default settings, and only changed the template to “Print”.
